### DISCUSSION TITLE
Access Khepri cluster from non-Khepri nodes

### DISCUSSION THREAD

**Discussion by verbit** [rabbitmq/khepri#221] [Q&A] [answered]
Is there a way to access a Khepri cluster via the `khepri` API _without_ running a Khepri store? Use case is the following: I have a cluster of 10 nodes and I want to run a Khepri cluster only on a subset of those nodes (3) but access it from all the machines.

**Comment by dumbbell** [comment, marked as answer]
Hi!

Currently no. You would have to either start a local instance of the store, or use `erpc:call()`.

This is something we could consider, but not right now as there are other priorities for our team. Is this something you would like to contribute?

**Reply by verbit** [reply]
Alright, thanks!

> Is this something you would like to contribute?

I'm not well experienced in Erlang, so it depends on whether it's a "good first issue" or not 😉. 

IIUC, `ra` actually supports calling commands on remote nodes. So `khepri` would need something along those lines as well. From glancing over the code base, it looked like it would be possible to extend (in a backwards-compatible manner) `StoreId` to take a shape like `{atom(), node()}` and adjust the dependant code (`process_command` and `process_query` in particular). But I might be vastly underestimating the work to be done.

**Reply by dumbbell** [reply]
Yes, this sounds like a good approach for the implementation! From an API usability point of view, this is fine as well.
